packages/docs/Examples/sale.md

```diff
@@ -159,24 +283,70 @@ await bob.broadcast(finalTx)
 
 ## Ordinals Sale
 
-The `Sale` smart contract does not preserver ordinal ranges. That is, the ordinal range of the NFT before the sale will not be the same as the ordinal range after the swap.
+The `Sale` smart contract is not safe to use with ordinals because the smart objects have different ordinal ranges before and after the call. To preserve the ordinal ranges the expression must not use the `_amount` keyword and must not return an object or an array containing an object.
+
+Building a sale contract for ordinals is more complicated than for smart objects. A very clever construction was proposed by Rodarmor [here](https://github.com/ordinals/ord/issues/802) and later [refined](https://github.com/ordinals/ord/issues/802#issuecomment-1498030294). Our smart contract below implements this exact idea. 
+
 
-### Smart Contract 
+### Smart Contracts
+
+The `exec` function of the `OrdSale` class swaps the owners just like in the `Sale` class. However it then proceeds to double the amount of `b1` and return `[b1, t, p, b2]`.
 
 ```ts
-export class Sale extends Contract {
-  static exec(b1: Valuable, b2: Valuable, t: NFT, p: NFT) {
-    const [ownerT] = t._owners
+class OrdSale extends Contract {
+  static exec(b1: Payment, b2: Payment, n: NFT, p: Payment) {
+    const [ownerT] = n._owners
     const [ownerP] = p._owners
-    t.transfer(ownerP)
+    n.transfer(ownerP)
     p.transfer(ownerT)
+
     b1.setAmount(b1._amount + b2._amount)
-    return [b1, t, p, b2]
+    return [b1, n, p, b2]
   }
 }
 ```
 
-### Usage
+When the `exec` function is evaluated, a transaction of the following form is built:
+
+```
+Inputs: b1, b2, t, p
+Outputs: b1, t, p, b2
+```
+
+The amount of `b1` after the call is the sum of the amounts of `b1` and `b2` before combined. The smart object `b1` therefor absorbs the entire ordinals range of `b2`. The objects `n` and `p` do not change their amounts during the call, therefore these objects preserver their ordinal ranges.
```

Comment:
Typo: The smart object `b1` **therefore**...
Typo: ...therefore these objects **preserve** their ordinal ranges.